files/job_wrapper.sh

```diff
@@ -41,76 +50,128 @@ find_module(){
-        container_path=${found_list[0]}
+        CONTAINER_PATH=${found_list[0]}
         return 0
     else
         return 1 # No containers found
     fi
 }
 
+# Setting up cmd line args for singularity
+# Print's stdout the argument line for running singularity utilizing
+setup_singularity(){
+
+    # Send cache to $SCRATCH or to the condor scratch directory
+    # otherwise
+    if [ -z "$SCRATCH" ]; then
+        export SINGULARITY_CACHEDIR="$_CONDOR_SCRATCH_DIR"
+    else
+        export SINGULARITY_CACHEDIR="$SCRATCH"
+    fi
+echo "exec -B $REANA_WORKFLOW_DIR:/reana docker://$DOCKER_IMG"
+}
+
+# Setting up shifter. Pull the docker_img into the shifter image gateway
+# and dump required arguments into stdout to be collected by a function call
+setup_shifter(){
+    # Check for shifterimg
+    if [[ ! $(command -v shifterimg 2>/dev/null) ]]; then
+        echo "Error: shifterimg not found..." >&2
+        exit 127
+    fi
+
+    # Attempt to pull image into image-gateway
+    if ! shifterimg pull "$DOCKER_IMG" >/dev/null 2>&1; then
+        echo "Error: Could not pull img: $DOCKER_IMG" >&2 
+        exit 127
+    fi
+
+    # Put arguments into stdout to collect
+    echo "--image=docker:${DOCKER_IMG} --volume=${REANA_WORKFLOW_DIR}:/reana -- "
+}
+
+# Setting up the arguments to pass to a container technology.
+# Currently able to setup: Singularity and Shifter.
+# Creates cmd line arguements for containers and pull image if needed (shifter)
+# Global arguments is used as the arguments to a container
+setup_container(){
+    # Need to cleanup to make more automated.
+    # i.e. run through the same list in find_container
+    local container=$(basename "$CONTAINER_PATH")
+
+    if [ "$container" == "singularity" ]; then
+        cntr_arguments=$(setup_singularity)
+    elif [ "$container" == "shifter" ]; then
+        cntr_arguments=$(setup_shifter)
+    else
+        echo "Error: Unrecognized container: $(basename $CONTAINER_PATH)" >&2
+        exit 127
+    fi
+}
+
 ######## Setup environment #############
 # @TODO: This should be done in a prologue
 # in condor via +PreCmd, eventually.
 #############################
-# Send cache to $SCRATCH or to the condor scratch directory
-# otherwise
-if [ "x$SCRATCH" == "x" ]; then
-    export SINGULARITY_CACHEDIR="$_CONDOR_SCRATCH_DIR"
-else
-    export SINGULARITY_CACHEDIR="$SCRATCH"
-fi
 
 find_container
 if [ $? != 0 ]; then
     echo "[Error]: Container technology could not be found in the sytem." >&2
     exit 127
 fi
 populate
+setup_container
 
 ######## Execution ##########
-# exec "$singularity_path" "$@"
 # Note: Double quoted arguments are broken
 # and passed as multiple arguments
 # in bash for some reason, working that
 # around by dumping command to a
-# temporary wrapper file.
+# temporary wrapper file named tmpjob.
 tmpjob=$(mktemp -p .)
 chmod +x $tmpjob 
-echo "$container_path" "$@" > $tmpjob
+echo "$CONTAINER_PATH" "$cntr_arguments" "bash -c \"cd /reana; ${@:3}\"" > $tmpjob
```

Comment:
I think reana already does `bash -c \"cd /reana; ` when setting `self.cmd`. You should be able to see that doing: condor_q -af Args (Arguments?) on submitted jobs.